### DISCUSSION TITLE
error: No type assignment with name `foo` found in the module `bar`

### DISCUSSION THREAD

**Discussion by ThibFrgsGmz** [esa/asn1scc#245] [Q&A] [answered]
Hi,

I am trying to factor my types as much as possible in my `ASN.1` definition and I have an error that some type definitions are missing.

I define a simple `Commands.asn` file:

```asn1
COMMAND DEFINITIONS::=
BEGIN

    CMD-HEADER ::= SEQUENCE {
        arg1 INTEGER(1 .. 4),
        arg2 INTEGER(1 .. 3)
    }

    CMD-STANDBY ::= SEQUENCE {
        header CMD-HEADER
    }
END
```

The `asn1scc` compiler gives me the error `error: No type assignment with name 'CMD-HEADER' found in the module 'COMMANDS'`.

Do you have any idea how to solve this problem?

I saw that there was no problem if I defined the `CMD-HEADER` object in another file and imported it into the `COMMANDS` module.

**Comment by maxime-esa** [comment]
I can't reproduce. Can you attach the file that is causing the issue? 
The error you report does not match the model you provide (there is no module `COMMANDS` in your example)

Also please provide the command line you used.
I tried the following:
```
$ asn1scc -c Commands.asn
```
and I got no errors.

**Comment by ThibFrgsGmz** [comment, marked as answer]
Ok thanks I found the error thanks to your remark @maxime-esa.

I have two `Commands.asn` files in two different folders and each defines the `COMMANDS` module.

In my list of `ASN.1` files that I send to the compiler I had :

```sh
asn1files="$BASE_DIR/template/Commands.asn
           $BASE_DIR/template/Telemetry.asn
           $BASE_DIR/icd_foo/Commands.asn
           $BASE_DIR/icd_foo/Telemetry.asn"
```

The `CMD-HEADER` object is only defined in the `icd_foo/Commands.asn` file. So there must be some kind of module name conflict or something.

Now that I have changed the name of the second `COMMANDS` module, I no longer have the error.

I will be more careful to make sure that the name of each module is unique in the future!

PS: I don't know why there is a missing "S" in the snippet I shared, I thought I copied/pasted and not rewrote. 🤔
